### PR TITLE
Implement scheme switching

### PR DESCRIPTION
**Pull request by jatkinson1000**
This PR puts the infrastructure in place to run either `ZM` or `YOG`.
At present both options will run `ZM` code, with alternative calls for `YOG` code to be added in future PRs.

- Implement `YOG` as an option for `deep_scheme` in `convect_deep.F90`.

### CODE REVIEW

File: bld/namelist_files/namelist_definition.xml [hunk 1]
```diff
@@ -2955,12 +2955,31 @@ Default: 10
 <!-- Moist Convection and Microphysics -->
 
 <entry id="deep_scheme" type="char*16" category="conv"
-       group="phys_ctl_nl" valid_values="ZM,UNICON,off,CLUBB_SGS" >
+       group="phys_ctl_nl" valid_values="ZM,UNICON,off,CLUBB_SGS,YOG" >
 Type of deep convection scheme employed.  'ZM' for Zhang-McFarlane;
-'off' for none; or 'UNICON' which doesn't distinquish shallow and deep.
+'off' for none;'YOG' for Yanni-O'Gorman; or 'UNICON' which doesn't
```
Comment (dorchard) [inline bld/namelist_files/namelist_definition.xml]: Yuval-O'Gorman

Comment (mondus) [inline bld/namelist_files/namelist_definition.xml]:
```suggestion
'off' for none;'YOG' for Yuval-O'Gorman; or 'UNICON' which doesn't
```

Comment (jatkinson1000) [reply bld/namelist_files/namelist_definition.xml]: Done with @mondus' suggestion below in https://github.com/m2lines/CAM-ML/pull/7/commits/6c9c99c496d65afaf0faf13ad0314ea5f3a6d387

**Comment by jatkinson1000** [comment]
This PR puts the infrastructure in place to run either `ZM` or `YOG`.

At present both options will run `ZM` code, with alternative calls for `YOG` code to be added in future PRs.

**Comment by jatkinson1000** [comment]
Rebased on #2 after changes there.

**Comment by jatkinson1000** [comment]
Following discussions with @paogorman there are no longer any changes on top of those in #2 as `YOG` is now its own scheme rather than being a `deep_scheme` option.

This is following a misunderstanding early in the project with how and where to implement the scheme.

#9 now performs tasks to implement switching of YOG on and off within `physpkg.F90` and to read the specific options in to a `yog_namelist`.

The appropriate way to run the code moving forwards is to set `yog_scheme='on'` and `deep_scheme='off'` in the CAM namelist file.

As such I am closing this pull request.

**Pull request closed without merging**